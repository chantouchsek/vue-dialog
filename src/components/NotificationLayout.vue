<template>
  <transition name="vue-dialog-notification-fade" @after-leave="onTransitionEnd">
    <div
      :class="['vue-dialog-notification', horizontalClass]"
      v-show="isActive"
      :style="getStyle"
      @mouseenter="clearTimer"
      @mouseleave="startTimer"
      role="alert"
    >
      <dialog-child v-bind="$options.propsData" ref="dialog" />
      <div
        class="vue-dialog-notification__closeBtn"
        v-if="showClose"
        @click.stop="close"
        v-html="'Ã—'"
      />
    </div>
  </transition>
</template>

<script>
import Notifiable from '../mixins/notifiable'

export default {
  mixins: [Notifiable],
  props: {
    width: {
      type: Number,
      default: () => 500
    }
  },
  computed: {
    getStyle () {
      return {
        [this.verticalProperty]: `${this.verticalOffset}px`,
        'max-width': `${this.width}px`,
        'z-index': this.zIndex
      }
    }
  }
}
</script>

<style>
    .vue-dialog-notification {
    display: flex;
    box-sizing: border-box;
    position: fixed;
    box-shadow: 0 2px 12px 0 rgba(0,0,0,.1);
    transition: opacity .3s, transform .3s, left .3s, right .3s, top .4s, bottom .3s;
    overflow: hidden;
  }

  .vue-dialog-notification>div:first-child {
    width: 100%;
  }

  .vue-dialog-notification.right {
    right: 16px;
  }

  .vue-dialog-notification.left {
    left: 16px;
  }

  .vue-dialog-notification__closeBtn {
    position: absolute;
    top: 9px;
    right: 15px;
    cursor: pointer;
    color: #909399;
    font-size: 22px;
  }

  .vue-dialog-notification__closeBtn:hover {
    color:#606266;
  }

  .vue-dialog-notification-fade-enter.right{
    right: 0;
    transform: translateX(100%);
  }

  .vue-dialog-notification-fade-enter.left{
    left: 0;
    transform: translateX(-100%);
  }

  .vue-dialog-notification-fade-leave-active {
    opacity: 0;
  }

  @media screen and (max-width: 450px) {
    .vue-dialog-notification {
      left: 8px !important;
      right: 8px !important;
      max-width: inherit !important;
    }
  }
</style>
