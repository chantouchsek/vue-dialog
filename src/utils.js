export function destroyVueElement (vm) {
  if (vm && !vm._isDestroyed && (typeof vm.$destroy === 'function')) {
    vm.$destroy()
  }
}

export function findContainer (container) {
  let found
  if (typeof container === 'string') {
    found = document.querySelector(container)
  } else {
    found = container
  }
  if (!found) {
    found = document.body
  }
  return found
}

// todo
// export function middlewareSeries (promises, appContext) {
//   if (!promises.length || appContext._redirected || appContext._errored) {
//     return Promise.resolve()
//   }
//   return promisify(promises[0], appContext)
//     .then(() => {
//       return middlewareSeries(promises.slice(1), appContext)
//     })
// }
