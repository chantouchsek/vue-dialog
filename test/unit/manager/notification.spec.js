import Vue from 'vue'
import DialogManager from '../../../src/manager'
import NotificationLayout from '../../../src/components/NotificationLayout'
import { sleep } from '../../utils'

describe('manager', () => {
  let manager

  beforeAll(() => {
    manager = new DialogManager({ context: { store: {} } })
    expect(manager).toBeInstanceOf(DialogManager)
    manager.layout('notification', NotificationLayout)
    manager.component('notification', {
      template: '<div>{{ text }}</div>',
      layout: 'notification',
      props: ['text']
    }, {
      waitForResult: true
    })
  })

  test('Test success notification', async () => {
    const dlg = await manager.notification({
      type: 'success',
      text: 'Test notification',
      timeout: 100,
      waitForResult: false
    })
    await sleep(50)
    expect(dlg.vm.$el).toMatchSnapshot()
    await sleep(100)
    await Vue.nextTick()
    expect(document.body.innerHTML).toBe('')
  })

  test('Test error notification with top-left position', async () => {
    const dlg = await manager.notification({
      type: 'error',
      text: 'Test notification',
      timeout: 100,
      position: 'top-left',
      waitForResult: false
    })
    await sleep(50)
    expect(dlg.vm.$el).toMatchSnapshot()
    await sleep(500)
    await Vue.nextTick()
    expect(document.body.innerHTML).toBe('')
  })

  test('Test warning notification with bottom-right position', async () => {
    const dlg = await manager.notification({
      type: 'warning',
      text: 'Test notification',
      timeout: 100,
      position: 'bottom-right',
      waitForResult: false
    })
    await sleep(50)
    expect(dlg.vm.$el).toMatchSnapshot()
    await sleep(1000)
    await Vue.nextTick()
    expect(document.body.innerHTML).toBe('')
  })
})
