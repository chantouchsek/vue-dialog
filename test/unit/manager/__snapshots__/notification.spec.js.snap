// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`manager Test error notification with top-left position 1`] = `
<div
  class="vue-dialog-notification left"
  role="alert"
  style="max-width: 500px; z-index: 2000; top: 10px;"
>
  <div>
    Test notification
  </div>
   
  <div
    class="vue-dialog-notification__closeBtn"
  >
    ×
  </div>
</div>
`;

exports[`manager Test success notification 1`] = `
<div
  class="vue-dialog-notification right"
  role="alert"
  style="max-width: 500px; z-index: 2000; top: 10px;"
>
  <div>
    Test notification
  </div>
   
  <div
    class="vue-dialog-notification__closeBtn"
  >
    ×
  </div>
</div>
`;

exports[`manager Test warning notification with bottom-right position 1`] = `
<div
  class="vue-dialog-notification right"
  role="alert"
  style="max-width: 500px; z-index: 2000; bottom: 10px;"
>
  <div>
    Test notification
  </div>
   
  <div
    class="vue-dialog-notification__closeBtn"
  >
    ×
  </div>
</div>
`;
